src/IECore/MeshTangentsOp.cpp

```diff
@@ -393,12 +258,15 @@ void MeshTangentsOp::modifyTypedPrimitive( MeshPrimitive * mesh, const CompoundO
 
 	bool orthoTangents = orthogonalizeTangentsParameter()->getTypedValue();
 
-	CalculateTangents f( vertsPerFace->readable(), mesh->vertexIds()->readable(), uData->readable(), vData->readable(), uvIndicesData->readable(), orthoTangents );
-
-	despatchTypedData<CalculateTangents, TypeTraits::IsFloatVec3VectorTypedData, HandleErrors>( pData, f );
+	std::pair<IECore::PrimitiveVariable, IECore::PrimitiveVariable> tangentPrimVars = IECore::MeshAlgo::calculateTangents(
+		mesh,
+		uPrimVarName + vPrimVarName,
```

Comment:
I'm not sure this is so simple unfortunately. Those parameters could have been set to "map1_s" and "map1_t" for example, and in that case we want to pass "map1" here.